saltproc/app.py

```diff
@@ -264,47 +276,58 @@ def _create_simulation_object(simulation_input, depcode, cores, nodes):
 
 def _create_reactor_object(reactor_input):
     """Helper function for `run()` """
-    msr = Reactor(
-        volume=reactor_input['volume'],
-        mass_flowrate=reactor_input['mass_flowrate'],
-        power_levels=reactor_input['power_levels'],
-        dep_step_length_cumulative=reactor_input['dep_step_length_cumulative'])
+    msr = Reactor(**reactor_input)
     return msr
 
 
-def _process_main_input_reactor_params(reactor_input, num_depsteps):
+def _process_main_input_reactor_params(reactor_input, n_depletion_steps, codename):
     """
     Process SaltProc reactor class input parameters based on the value and
-    data type of the `num_depsteps` parameter, and throw errors if the input
+    data type of the `n_depletion_steps` parameter, and throw errors if the input
     parameters are incorrect.
     """
-    dep_step_length_cumulative = reactor_input['dep_step_length_cumulative']
+
+    depletion_timesteps = reactor_input['depletion_timesteps']
     power_levels = reactor_input['power_levels']
-    if num_depsteps is not None and len(dep_step_length_cumulative) == 1:
-        if num_depsteps < 0.0 or not int:
-            raise ValueError('Depletion step interval cannot be negative')
-        # Make `power_levels` and `dep_step_length_cumulative`
-        # lists of length `num_depsteps`
+
+    if n_depletion_steps is not None:
+        if n_depletion_steps < 0.0 or not int:
+            raise ValueError('There must be a positive integer number'
+                             ' of depletion steps. Provided'
+                             f' n_depletion_steps: {n_depletion_steps}')
+        if len(depletion_timesteps) == 1:
+            depletion_timesteps = depletion_timesteps * n_depletion_steps
+        if len(power_levels) == 1:
+            power_levels = power_levels * n_depletion_steps
+
+    if len(depletion_timesteps) != len(power_levels):
+        raise ValueError('depletion_timesteps and power_levels length mismatch:'
+                         f' depletion_timesteps has {len(depletion_timesteps)}'
+                         f' entries and power_levels has {len(power_levels)}'
+                         ' entries.')
+
+    if reactor_input['timestep_type'] == 'cumulative':
+        ts = list(np.diff(depletion_timesteps))
+        depletion_timesteps = depletion_timesteps[0] + ts
```

Comment:
```suggestion
        depletion_timesteps = np.cumsum(depletion_timesteps)
```

I agree with @LukeSeifert that the expected result is unclear. Possibly use ``np.cumsum()``?

Reply:
This is actually coded improperly now that I look at it. What we are supposed to be doing here is converting a cumulative timeseries (1s, 3s, 5s) to a stepwise timeseries (1s, 2s, 2s). But I eneterd this wrong which explains why we were confused. I seem to be missing tests for this function as well so I'll add those